transport/grpc/outbound.go

```diff
@@ -81,3 +92,24 @@ func callDownstream(
 	headers := applicationHeaders.fromMetadata(responseHeaders, transport.Headers{})
 	return &transport.Response{Body: closer, Headers: headers}, nil
 }
+
+func getErrFromGRPCError(err error, treq *transport.Request, ttl time.Duration) error {
+	switch grpc.Code(err) {
+	// TIMEOUT
+	case codes.DeadlineExceeded:
+		return errors.ClientTimeoutError(treq.Service, treq.Procedure, ttl)
+
+	// BAD REQUEST
+	case codes.Unimplemented, codes.InvalidArgument, codes.NotFound:
+		return errors.RemoteBadRequestError(grpc.ErrorDesc(err))
+
+	// UNEXPECTED
+	case codes.Canceled, codes.AlreadyExists, codes.PermissionDenied,
+		codes.Unauthenticated, codes.ResourceExhausted, codes.FailedPrecondition,
+		codes.Aborted, codes.OutOfRange, codes.Internal,
+		codes.Unavailable, codes.DataLoss, codes.Unknown:
+		fallthrough
```

Comment:
might be a good idea to enumerate them *shrug* we can remove them, it should be no problem